mm2src/mm2_main/src/lp_swap/swap_v2_rpcs.rs

```diff
@@ -501,3 +503,165 @@ pub(crate) async fn active_swaps_rpc(
+        result: "Success".to_string(),
+    })
+}
+
+#[derive(Deserialize)]
+pub(crate) struct KickStartSwapRequest {
+    uuid: Uuid,
+}
+
+#[derive(Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub(crate) enum KickStartSwapErr {
+    Internal(String),
+    AlreadyRunning,
+    AlreadyFinished,
+    NeedsCoinActivation(String),
+    NotFound,
+}
+
+impl HttpStatusCode for KickStartSwapErr {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            KickStartSwapErr::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            KickStartSwapErr::AlreadyRunning
+            | KickStartSwapErr::AlreadyFinished
+            | KickStartSwapErr::NeedsCoinActivation(_) => StatusCode::BAD_REQUEST,
+            KickStartSwapErr::NotFound => StatusCode::NOT_FOUND,
+        }
+    }
+}
+
+#[derive(Serialize)]
+pub(crate) struct KickStartSwapResponse {
+    result: String,
+}
+
+pub(crate) async fn kickstart_swap_rpc(
+    ctx: MmArc,
+    req: KickStartSwapRequest,
+) -> MmResult<KickStartSwapResponse, KickStartSwapErr> {
+    // Make sure first that the swap isn't already running. Note that this is not atomic and we might still end
+    // up with the same swap being kickstarted twice, but we have filesystem swap locks for that. This check is
+    // rather for convenience.
+    let swap_ctx = SwapsContext::from_ctx(&ctx).map_err(KickStartSwapErr::Internal)?;
+    if swap_ctx.running_swaps.lock().unwrap().contains_key(&req.uuid) {
+        return MmError::err(KickStartSwapErr::AlreadyRunning);
+    }
+    // Load the swap from the DB.
+    let swap = match SavedSwap::load_my_swap_from_db(&ctx, req.uuid).await {
+        Ok(Some(s)) => s,
+        Ok(None) => {
+            return MmError::err(KickStartSwapErr::NotFound);
+        },
+        Err(e) => {
+            return MmError::err(KickStartSwapErr::Internal(format!(
+                "Error getting the swap from the DB: {e}"
+            )))
+        },
+    };
+    // Make sure that the swap isn't finished.
+    if swap.is_finished() {
+        return MmError::err(KickStartSwapErr::AlreadyFinished);
+    }
+    let maker_coin = match swap.maker_coin_ticker() {
+        Ok(coin) => match lp_coinfind(&ctx, &coin).await {
+            Ok(Some(coin)) => coin,
+            Ok(None) => {
+                return MmError::err(KickStartSwapErr::NeedsCoinActivation(format!(
+                    "Maker coin {} must be activated",
+                    coin
+                )));
+            },
+            Err(e) => {
+                return MmError::err(KickStartSwapErr::Internal(format!(
+                    "Error trying to find maker coin: {e}"
+                )))
+            },
+        },
+        Err(e) => {
+            return MmError::err(KickStartSwapErr::Internal(format!(
+                "Error getting maker ticker of swap: {e}"
+            )));
+        },
+    };
+    let taker_coin = match swap.taker_coin_ticker() {
```

Comment:
Code repetition for finding maker_coin and taker_coin.
(BTW could we just use lp_coinfind_or_err in all such cases?)

Reply:
thanks. i didn't like it either but needed some assertion xD

Reply:
[cc0db2e](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2280/commits/cc0db2ed4a038bffc5bdfdb0cbcf29b5b4204731)